flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/utils/UserDefinedFunctionUtils.scala

```diff
@@ -140,6 +147,25 @@ object UserDefinedFunctionUtils {
           s"one method named 'eval' which is public, not abstract and " +
           s"(in case of table functions) not static.")
     } else {
+      var trailingSeq = false
+      var trailingArray = false
+      methods.foreach(method => {
+        val signatures = method.getParameterTypes
+        if (signatures.nonEmpty) {
+          val trailingArg = signatures(signatures.length - 1)
+          if (trailingArg.getName.equals("scala.collection.Seq")) {
+            trailingSeq = true
+          } else if (trailingArg.isArray) {
+            trailingArray = true
+          }
+        }
+      })
+      if (trailingSeq && !trailingArray) {
```

Comment:
The `eval` method can be overloaded. Such as : 
```scala
def eval(args: Array[Int]): Int = {}
def eval(args: Int*): Int = {}
```
In this case, the methods can satisfy the condition, but the exception should be thrown.